### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by ger619**
**Create models**

- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.

**Use models to insert data**

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:

```
   first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
```

Create at least 4 posts written by one of the users you created by running the following code:

```
   first_post = Post.create(author_id: first_user, title: 'Hello', text: 'This is my first post')
```

Create at least 6 posts comments for one of the posts you created by running the following code

```
Comment.create(post: first_post, author_id: second_user, text: 'Hi Tom!' )
```

- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

**Create custom methods**
Your models should include some additional methods.
- User
  - A method that returns the 3 most recent posts for a given user 
- Post
  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.
- Comment
  - A method that updates the comments counter for a post.
- Likes
  - A method that updates the likes counter for a post.
- Go to `rails c` and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  has_many :comment
+  has_many :post
+  has_many :like
+end
```
Comment (phelian23) [inline app/models/user.rb:11]: - [x] As requested by the project requirements, the user model should have a method that returns the 3 most recent posts. Kindly add that.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comment
+  has_many :like
```
Comment (phelian23) [inline app/models/post.rb:4]: - [x] As requested by the project requirements, the Post model should have a method that updates the posts counter for a user
- [x] Also the Post model should have a method that returns the 5 most recent comments for a given post. Kindly add this methods

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+end
```
Comment (phelian23) [inline app/models/comment.rb:9]: - [ ] As requested by the project requirements, the Comment model should have a method that updates the comments counter for a post. Kindly add this.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+end
```
Comment (phelian23) [inline app/models/like.rb:9]: - [ ]  As requested by the project requirements, the Like model should have a method that updates the likes counter for a post. Kindly add this.

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @ger619,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 4 above)
Comment (ger619) [reply app/models/like.rb:9]: Thanks

**Comment by ger619** [comment]
@phelian23 done

**Review by Deepakdanger — APPROVED** [review]
Hi @ger619 , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**